### PR TITLE
Adding IfcEarthingElement/IfcEarthingElementType as new subtypes of IfcFlowTerminal/IfcFlowTerminalType

### LINKED ISSUE #403: [IFC Tunnel] Add new IfcEarthingElement incl Predefined types
Proposal:
- Added IfcEarthingElement under IfcFlowTerminal
- Added required PDTs

_Originally posted by @larswik in https://github.com/bSI-InfraRoom/IFC-Specification/issues/203#issuecomment-1094977138_

Entity IfcEarthingElement : Subtype of IfcFlowTerminal
PDTs:
EARTHINGSTRIP : According to EN 62561-2, for use in earth-termination and lightning protection systems as well as for ring equipotential bonding.
GROUNDINGPLATE : IEC 62561-2 Earth electrode consisting of a metal plate buried in the ground
GROUNDINGROD : IEC 62561-2 Earth electrode consisting of a metal rod driven into the ground
FIXEDTERMINAL : According to EN 62561-2, for installation in concrete as a corrosion-free connection to the earth-termination system for protective equipotential bonding and / or functional equipotential bonding of the down conductor
GROUNDINGMESH : IEC 60050-531 An electrode in the form of a mesh, and designed to establish an equipotential plan.

### PR DESCRIPTION
**Pull request by larswik**
Fixes #403

### CODE REVIEW

File: IFC4x3/Constants/g/GROUNDINGMESH_1WFsqjqs54qPDn6VzU$xcd.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="GROUNDINGMESH_1WFsqjqs54qPDn6VzUxcd" Name="GROUNDINGMESH" UniqueId="603f6d2d-d361-44d1-9371-19ff5effb9a7">
+	<Documentation>IEC 60050-531 An electrode in the form of a mesh, and designed to establish an equipotential plan.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/g/GROUNDINGMESH_1WFsqjqs54qPDn6VzU$xcd.xml]:
```suggestion
	<Documentation>Definition according to IEC 60050-531: An electrode in the form of a mesh, and designed to establish an equipotential plan.</Documentation>
```

File: IFC4x3/Constants/f/FIXEDTERMINAL_3HhDuEn2f2nP6ZL0SQeoy3.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="FIXEDTERMINAL_3HhDuEn2f2nP6ZL0SQeoy3" Name="FIXEDTERMINAL" UniqueId="d1acde0e-c42a-42c5-91a3-54071aa32f03">
+	<Documentation>According to EN 62561-2, for installation in concrete as a corrosion-free connection to the earth-termination system for protective equipotential bonding and / or functional equipotential bonding of the down conductor</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/f/FIXEDTERMINAL_3HhDuEn2f2nP6ZL0SQeoy3.xml]:
```suggestion
	<Documentation>Definition according to EN 62561-2: for installation in concrete as a corrosion-free connection to the earth-termination system for protective equipotential bonding and / or functional equipotential bonding of the down conductor</Documentation>
```

File: IFC4x3/Constants/e/EARTHINGSTRIP_3Fh7NdPNTF3fo7Avf2LN6_.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="EARTHINGSTRIP_3Fh7NdPNTF3fo7Avf2LN6_" Name="EARTHINGSTRIP" UniqueId="cfac75e7-6577-4f0e-9c87-2b9a425571be">
+	<Documentation>According to EN 62561-2, for use in earth-termination and lightning protection systems as well as for ring equipotential bonding.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/e/EARTHINGSTRIP_3Fh7NdPNTF3fo7Avf2LN6_.xml]:
```suggestion
	<Documentation>Definition according to EN 62561-2: for use in earth-termination and lightning protection systems as well as for ring equipotential bonding.</Documentation>
```

File: IFC4x3/Constants/g/GROUNDINGPLATE_3lDZcwSjL5iAoe6FsC_G2e.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="GROUNDINGPLATE_3lDZcwSjL5iAoe6FsC_G2e" Name="GROUNDINGPLATE" UniqueId="ef3639ba-72d5-45b0-aca8-18fd8cf900a8">
+	<Documentation>IEC 62561-2 Earth electrode consisting of a metal plate buried in the ground</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/g/GROUNDINGPLATE_3lDZcwSjL5iAoe6FsC_G2e.xml]:
```suggestion
	<Documentation>Definition according to IEC 62561-2: Earth electrode consisting of a metal plate buried in the ground.</Documentation>
```

File: IFC4x3/Constants/g/GROUNDINGROD_25zpcaPd9AvhSIrZdgJ$rm.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="GROUNDINGROD_25zpcaPd9AvhSIrZdgJrm" Name="GROUNDINGROD" UniqueId="85f739a4-6672-4ae6-b712-d639ea4ffd70">
+	<Documentation>IEC 62561-2 Earth electrode consisting of a metal rod driven into the ground</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/g/GROUNDINGROD_25zpcaPd9AvhSIrZdgJ$rm.xml]:
```suggestion
	<Documentation>Definition according to IEC 62561-2: Earth electrode consisting of a metal rod driven into the ground.</Documentation>
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcElectricalDomain/Entities/IfcEarthingElementType/DocEntity.xml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcEarthingElementType" UniqueId="17dc2a6c-8e78-43ad-88d3-ac423768d8cf" BaseDefinition="IfcFlowTerminalType" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="b915dbe1-d893-418a-a3fb-3575a00a4e39" DefinedType="IfcEarthingElementTypeEnum" AttributeFlags="1">
+			<Documentation>Identifies the predefined types of earting element from which the type required may be set.</Documentation>
+		</DocAttribute>
+	</Attributes>
+	<WhereRules>
+		<DocWhereRule Name="CorrectPredefinedType" UniqueId="50650d61-371b-4e09-9af1-f9d9d2880c25">
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcElectricalDomain/Entities/IfcEarthingElementType/DocEntity.xml:9]: Could add documentation for WRs.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcElectricalDomain/Entities/IfcEarthingElement/DocEntity.xml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcEarthingElement" UniqueId="c8d5b601-4a86-4bda-a5ba-3113204765da" BaseDefinition="IfcFlowTerminal" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="0c8d7a95-fbfc-429e-8d05-3358757d299d" DefinedType="IfcEarthingElementTypeEnum" AttributeFlags="1">
+			<Documentation>Identifies the predefined types of earting element from which the type required may be set.</Documentation>
+		</DocAttribute>
+	</Attributes>
+	<WhereRules>
+		<DocWhereRule Name="CorrectPredefinedType" UniqueId="8d91aa6e-3896-452b-9020-199ec5746ba6">
+			<Expression>NOT(EXISTS(PredefinedType)) OR
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcElectricalDomain/Entities/IfcEarthingElement/DocEntity.xml:11]: Documentation for WRs.

**Review by SergejMuhic — CHANGES_REQUESTED** [review]
1. Adjusted the constant documentation to be in-line with the rest of IFC documentation
2. Could add the documentation for WRs

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcElectricalDomain/Entities/IfcEarthingElementType/DocEntity.xml (hunk 6 above)
Comment (larswik) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcElectricalDomain/Entities/IfcEarthingElementType/DocEntity.xml:9]: Done.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcElectricalDomain/Entities/IfcEarthingElement/DocEntity.xml (hunk 7 above)
Comment (larswik) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcElectricalDomain/Entities/IfcEarthingElement/DocEntity.xml:11]: Done.

**Review by anborr — APPROVED** [review]

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**